controllers/util/misc.go

```diff
@@ -72,6 +72,24 @@ func AddLabel(obj client.Object, key, value string) bool {
 	return !labelAdded
 }
 
+func AddAnnotation(obj client.Object, key, value string) bool {
+	const added = true
+
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+
+	if _, ok := annotations[key]; !ok {
```

Comment:
It is good to prevent making unnecessary updates, but what if the value is not the same? Should we test here whether `annotations[key] != value`?

Reply:
Addressed, better to change the value if so desired.